flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/common/CommonExecTableSourceScan.java

```diff
@@ -103,56 +106,63 @@ protected Transformation<RowData> translateToPlanInternal(
         final ScanTableSource tableSource =
                 tableSourceSpec.getScanTableSource(
                         planner.getFlinkContext(), ShortcutUtils.unwrapTypeFactory(planner));
+        final String sourceInfo =
+                tableSourceSpec.getContextResolvedTable().getIdentifier().asSummaryString();
```

Comment:
For those tableSourceSpec that points to the same source, will they have the same `ScanTableSource.ScanRuntimeProvider provider` instance?

If yes, instead of adding an extra interface `WithSourceInfo` to maintain the source table identifier in selected Transformation subclass, would it be more intuitive to have the same instance of `ScanRuntimeProvider` always return the same Transformation, similar to what is done for `TransformationScanProvider`?

Reply:
Thanks @lindong28 . I've updated the PR, could you help review it again?